docker/app_dev

```diff
@@ -1,33 +1,34 @@
-# Application Continuous Integration image
-# Contains a CCF release, compile toolchain and Azure CLI tooling
+# Application Development image
+# Contains a CCF release for platform and toolchain for target platform
 
-FROM ubuntu:20.04
+ARG platform=sgx
 
-ARG extra_vars
+# SGX
+FROM ubuntu:20.04 AS base-sgx
 
-RUN echo "APT::Acquire::Retries \"5\";" | tee /etc/apt/apt.conf.d/80-retries
+WORKDIR /
+COPY ./docker/sgx_deps_pin.sh /
+RUN ./sgx_deps_pin.sh && rm /sgx_deps_pin.sh
+
+# SNP
+FROM ubuntu:20.04 AS base-snp
 
-# Work-around for https://github.com/intel/linux-sgx/issues/395
-RUN mkdir -p /etc/init
+# Virtual
+FROM ubuntu:20.04 AS base-virtual
 
-ENV UBUNTU=focal
-ENV PSW_VERSION=2.17.100
-RUN if [ -z "$PSW_VERSION" ]; then echo "Please set PSW_VERSION (e.g. 2.11)." >&2; exit 1; fi
+# Final dev image
+FROM base-${platform} AS final
 
-RUN apt-get update && apt-get install -y wget gnupg
+ARG platform=sgx
```

Comment:
Do we need this a second time?

Reply:
Unfortunately yes as the build arg needs to be repeated for all stages (it took me a while to find out it was necessary!).